### DISCUSSION TITLE
Allure step fail is not showing test as fail

### DISCUSSION THREAD

**Discussion by krishtoautomate** [allure-framework/allure2#2546] [Questions & Support] [answered]
### Describe the Bug

When step is marked as fail using below code, the test is still marked as pass.

`Allure.step(message, io.qameta.allure.model.Status.FAILED);`

**actual:**

<img width="1342" alt="image" src="https://github.com/allure-framework/allure2/assets/20734341/4c576b1b-3e21-484e-84ed-e2c58eb7e445">

**expected:** 
<img width="1352" alt="image" src="https://github.com/allure-framework/allure2/assets/20734341/f4fba714-8642-44b1-9a08-57ae0eb02443">

### Steps to Reproduce

add below step to java code:

`Allure.step(message, io.qameta.allure.model.Status.FAILED);`

### Expected Behaviour

test should be marked as fail as its step is marked to fail

### What Language are you using?

Java

### What Framework/Allure Integration you are using?

allure-testng

### What version of Allure Integration you are using?

2.26.0

### What version of Allure Report you are using?

2.26.0

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by baev** [comment, marked as answer]
Allure steps are not changing the way the test is executed. Basically, it's just a **structured logging**. The test result status in the report represents the status provided by the test framework.
